extensions/resteasy-reactive/quarkus-resteasy-reactive-links/deployment/src/main/java/io/quarkus/resteasy/reactive/links/deployment/LinksProcessor.java

```diff
@@ -194,6 +197,52 @@ private RuntimeValue<GetterAccessorsContainer> implementPathParameterValueGetter
         return getterAccessorsContainer;
     }
 
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateClassHasFieldId(IndexView index, String entityType) {
+        // create a new independent class name that we can override
+        DotName className = DotName.createSimple(entityType);
+        ClassInfo classInfo = index.getClassByName(className);
+
+        if (classInfo != null) {
+            validateRec(index, entityType, classInfo);
+        }
+    }
+
+    /**
+     * Validates if the given classname contains a field `id` or annotated with `@Id`
+     *
+     * @throws IllegalStateException if the classname does not contain any sort of field identifier
+     */
+    private void validateRec(IndexView index, String entityType, ClassInfo classInfo) {
+        List<FieldInfo> fieldsNamedId = classInfo.fields().stream()
+                .filter(f -> f.name().equals("id"))
+                .collect(Collectors.toList());
+        List<AnnotationInstance> fieldsAnnotatedWithId = classInfo.annotations().stream()
+                .filter(a -> a.name().toString().endsWith("persistence.Id"))
+                .collect(Collectors.toList());
+
+        // Id field found, break the loop
+        if (!fieldsNamedId.isEmpty() || !fieldsAnnotatedWithId.isEmpty())
+            return;
+
+        // Id field not found and hope is gone
+        DotName superClassName = classInfo.superName();
+        if (fieldsNamedId.isEmpty() && fieldsAnnotatedWithId.isEmpty() && superClassName == null) {
+            throw new IllegalStateException("Cannot generate web links for the class " + entityType +
+                    " because is either missing an `id` field or a field with an `@Id` annotation");
+        }
+
+        // Id field not found but there's still hope
+        classInfo = index.getClassByName(superClassName);
+        if (superClassName != null && classInfo != null) {
```

Comment:
So, what happens if we have a `superClassName` but the superclass is not indexed? No error?

Reply:
How can I test for that scenario?

Reply:
Indeed, nothing would happen in that scenario. I added a check and exception for this case.